src/sqlfluff/dialects/dialect_ansi.py

```diff
@@ -2000,7 +2000,14 @@ class CaseExpressionSegment(BaseSegment):
             Dedent,
             "END",
         ),
-        terminators=[Ref("CommaSegment"), Ref("BinaryOperatorGrammar")],
+        terminators=[
+            OneOf(
```

Comment:
I don't think this OneOf is needed, as the OneOf is implicit in the terminators handling

Reply:
Yep, you are right 👍 
I didn't try without `OneOf`, it's still retaining the same feature result.
Removed it